airflow/models/taskinstance.py

```diff
@@ -2074,11 +2074,13 @@ def render_k8s_pod_yaml(self) -> Optional[dict]:
         kube_config = KubeConfig()
         pod = PodGenerator.construct_pod(
             dag_id=self.dag_id,
+            run_id=self.run_id,
             task_id=self.task_id,
+            map_index=self.map_index,
+            date=None,
             pod_id=create_pod_id(self.dag_id, self.task_id),
             try_number=self.try_number,
             kube_image=kube_config.kube_image,
-            date=self.execution_date,
```

Comment:
Not sure about compatibility here, otherwise lgtm.

Reply:
Should be okay, as we only include the label if it is gte 0

https://github.com/apache/airflow/pull/21871/files#diff-01764c9ba7b2270764a59e7ff281c95809071b1f801170ee75a02481a8a730aaR359-R360


I still think I'm missing some a case --  I haven't made a change to the matching "pod is finished" side of things to get the TI by map_index...

Reply:
Ah, added in https://github.com/apache/airflow/pull/21871/commits/d25926e711bf04e5fb2255a2db767ce6423c3c9c